backend/lib/utils.js

```diff
@@ -17,8 +19,32 @@ const bool = (env) => env == "true";
 
 const getCookieToken = (req) => req.cookies.token;
 
-const emailRegEx = /^\w+([.-]?\w+)*@\w+([.-]?\w+)*(\.\w{2,3})+$/;
-const isValidEmail = (email) => emailRegEx.test(email);
+// email validation code from front-end 'src\utils\validators'
+const isValidTopLevelDomain = (string) => {
+  return tlds.some(tld => string.endsWith("." + tld));
+};
+
+const isValidEmail = (email) => {
+  return _isEmail(email) && isValidTopLevelDomain(email);
+};
+
+// password validation code from front-end 'src\utils\validators'
+const SPECIAL_CHARS = /[!@#$%^&*.;]/;
+const containsLowerCase = (str) => /[a-z]/.test(str);
+const containsUpperCase = (str) => /[A-Z]/.test(str);
+const containsNumber = (str) => /\d/.test(str);
+const containsSpecialChars = (str) => SPECIAL_CHARS.test(str);
+
+const isValidPassword = (password) => {
+  const results = [
+    containsLowerCase(password),
+    containsUpperCase(password),
+    containsNumber(password),
+    containsSpecialChars(password),
+  ];
+
+  return results.filter((res) => res).length >= 3;
```

Comment:
could we rename the `results` to something else, also the flag `res` to something else? It is not as straight forward. Also do we want the password only to meet 3 out of 4 of those validators?

Reply:
how about `validCombo` and `valid`, while this code was copied from our front end code, may need the same update there. I am not sure how strict we want the password to be, this applies to both our F/E and B/E, @joshmorel has a ticket talking about if we should keep validation on both side #1641

Reply:
Hi @idjevm I renamed it to be `checks` and `check`, let me know if that works : )